### PR TITLE
Named Function Ref/Lookup

### PR DESCRIPTION
**Pull request by mtesseracttech**
Adds named function references and also fixes a bug where tests that stored their query in a CDATA block did not get loaded in properly.

The addition of the CDATA fix did not really get all that many more tests to pass, but it has changed the tests that failed on an empty query to ones that had a more meaningful query.

### CODE REVIEW

File: FontoXPathCSharp/EvaluationUtils/Atomize.cs [hunk 1]
```diff
@@ -77,7 +77,7 @@ public static ISequence AtomizeSequence<TNode>(ISequence sequence, ExecutionPara
                 if (domfacade.IsComment(node) || domfacade.IsProcessingInstruction(node))
                     return;
 
-                if (domfacade.IsText(node))
+                if (domfacade.IsText(node) || domfacade.IsCharacterData(node))
```
Comment (DrRataplan) [inline FontoXPathCSharp/EvaluationUtils/Atomize.cs:80]: the isText is true for all isCharacterData. So you can just use isCharacterData
